machine/v1/types_awsplacementgroup.go

```diff
@@ -0,0 +1,155 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AWSPlacementGroup ensures that a placement group matching the given configuration exists within AWS
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".spec.groupType",description="Placement Group Type"
+// +kubebuilder:printcolumn:name="Management",type="string",JSONPath=".spec.managementState",description="Management State"
+// +kubebuilder:printcolumn:name="Replicas",type="integer",JSONPath=".status.replicas",description="EC2 Replicas within the Placement Group"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="AWSPlacementGroup age"
+type AWSPlacementGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AWSPlacementGroupSpec   `json:"spec,omitempty"`
+	Status AWSPlacementGroupStatus `json:"status,omitempty"`
+}
+
+type AWSPlacementGroupSpec struct {
+	// ManagementState determines whether the placement group is expected
+	// to be managed by this CRD or whether it is user managed.
+	// A managed placement group may be moved to unmanaged, however an unmanaged
+	// group may not be moved back to managed.
+	// +kubebuilder:default:=Managed
+	// +optional
+	ManagementState ManagementState `json:"managementState"`
```

Comment:
per slack, this and awsplacementgroupunion for a discriminated union distinct from credentialsSecret

Reply:
Agreed, that makes sense, this is now pushed